polkadot-parachains/parachains-common/src/impls.rs

```diff
@@ -103,9 +103,16 @@ pub struct AssetsFrom<T>(PhantomData<T>);
 impl<T: Get<MultiLocation>> FilterAssetLocation for AssetsFrom<T> {
 	fn filter_asset_location(asset: &MultiAsset, origin: &MultiLocation) -> bool {
 		let loc = T::get();
+		log::trace!(target: "xcm::AssetsFrom", "loc: {:?}, origin: {:?}", loc, origin);
 		&loc == origin &&
-			matches!(asset, MultiAsset { id: AssetId::Concrete(asset_loc), fun: Fungible(_a) }
-			if asset_loc.match_and_split(&loc).is_some())
+			match asset {
+				MultiAsset { id: AssetId::Concrete(asset_loc), fun: Fungible(_a) } =>
+					loc.parent_count() == asset_loc.parent_count() &&
+						loc.interior().iter().zip(asset_loc.interior().iter()).all(
+							|(prefix_junction, asset_junction)| prefix_junction == asset_junction,
```

Comment:
if asset_loc.interior() is longer, then I think std iter zip will ignore any extra length. Is this intended?

Reply:
Should be fixed now.